indico/util/user.py

```diff
@@ -88,6 +90,12 @@ def principal_from_fossil(fossil, allow_pending=False, allow_groups=True, allow_
         if group.group is None and not allow_missing_groups:
             raise ValueError('Multipass group does not exist: {}:{}'.format(provider, id_))
         return group
+    elif allow_category_roles and event.category.roles:
```

Comment:
Instead of the `allow_*` flag  better add a `category` kwarg (and default to `event.category` when an event but no category is present).